test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.V2.json

```
@@ -30590,6 +30590,11 @@
             "type": "string"
           }
         },
+        "TripData": {},
```

Leave it an empty object? or use a 'type': 'object'?

---

```suggestion
        "TripData": { 
              "type": "object"
        },
```
@andrueastman Your thoughts?

---

We should leave it as empty. The reason being that the data could be a primitive as well(integer, string ...) or even a collection. As the type is only known at runtime so I think it would be incorrect to allude that it's an object.

---

Or we could have something like this (the list of types is not exhaustive): 
```
components:
  schemas:
    ExampleResource:
      type: object
      properties:
        id:
          type: string
        customData:
          anyOf:
            - type: string
            - type: number
            - type: integer
            - type: boolean
            - type: object
            - type: array
              items:
                anyOf:
                  - type: string
                  - type: number
                  - type: integer
                  - type: boolean
                  - type: object
```

---

Its not practical to model out everything it could be, as it could be a collection of structs each with a heterogeneous collection that's a mixture of bools, ints and more structs. So, as @andrueastman says, it's completely unknown at design time.

My JSON schema skill isn't strong, but whatever expresses that complete lack of knowledge whether it is primitive, object or array, that is what we should do.

---

Thanks, @garethj-msft